listener.go

```diff
@@ -132,18 +132,34 @@ func (l *listener) Close() error {
 	}
 	l.closed = true
 	err := l.Listener.Close()
-	// Drain nextConn and nextErr
-drain:
+
+	// Drain nextConn
+drainNextConn:
 	for {
 		select {
 		case conn := <-l.nextConn:
 			conn.Close()
-		case <-l.nextErr:
 		default:
-			break drain
+			break drainNextConn
 		}
 	}
 	close(l.nextConn)
+
+	// Drain nextErr
+drainNextErr:
+	for {
+		select {
+		case e := <-l.nextErr:
+			if e == nil { // it may already be closed
+				break drainNextErr
+			} else {
+				err = e
+			}
+		default:
+			break drainNextErr
+		}
+	}
+
```

Comment:
drive by fix, this can hang indefinitely if nextErr is closed before drain starts (will return nil forever).  Test was intermittently hanging before any changes.

Reply:
This may be able to be simplified a bit. PR at https://github.com/getlantern/cmux/pull/10

Reply:
oh nice, thanks

Reply:
@joesis Eh on second thought, this library has more complicated shutdown problems then I saw at first glance, I'm going to file this separately and just push though this error change as is ( https://app.zenhub.com/workspaces/lantern-55d6e412162fe7fc264ad9a8/issues/getlantern/lantern-internal/2888 )